### ISSUE TITLE
How to rewrite the configuration file for adding multi loss function modules to training workflow?

### ISSUE THREAD

**Issue by ter-li** [open-mmlab/mmpose#1347]
Hi, the mmpose group,

Thanks for the great work you open sourced and shared with the community. 

In my reuse implementations, I tried to create a new loss function and add it to the training workflow, however, I checked the documentation block as following and have trouble when rewrite the configuration file.

https://github.com/open-mmlab/mmpose/blob/a3814815835f4ca87f6f698ec8e837ecb119ec00/docs/zh_cn/tutorials/4_new_modules.md#L154-L214

There is a stupid question from me, **how to add multi registered loss function modules to the training workflow**, via rewriting the configuration(the following block) or any else?

https://github.com/open-mmlab/mmpose/blob/a3814815835f4ca87f6f698ec8e837ecb119ec00/docs/zh_cn/tutorials/4_new_modules.md#L210-L214

Looking forward for your reply.

Best,
Terry

**Comment by ly015** [comment]
Thanks for your interest. I am afraid you can not add multiple loss functions directly by rewriting the config file. Currently, the head module only supports a single loss function. However, you can implement multiple loss functions via minor refactoring, which shouldn't be much trouble. Here are 2 approaches to this:
1. Refactor the head module to take multiple loss function configs and build them during initialization. During training, the head should invoke all loss functions and collect the results.
2. Implement a 'loss wrapper`, which inherits the loss interface, but wrappers multiple actual loss functions. When the loss wrapper is invoked by the head, it calculates multiple wrapped losses and returns the collected results.

Please let us know if you need more help. Also, if you would like to implement a general solution to support multiple losses, welcome to contribute your implementation to MMPose :)

**Comment by ter-li** [comment]
Thanks for your rapid reply.

I used to try the second solution which implement several criterions in only one PyTorch module. But it seems hard to visualize the loss trend movements, which is the reason and motivation that I  raised an issue on whether I can add multi loss function modules when rewriting the configuration file.

To overcome the above issue, I think the first proposal is proper. But I need to say sorry. I am doing some urgent research staffs now and have no spare time to contribute. Maybe we can code together to enhance this after a while when I am free (maybe pass the NIPS/CVPR deadline :<). I will take a note and make it as one in my to-do list. :)

Best,
Terry
